maxMoviesInYear.js

```diff
@@ -1,30 +1,29 @@
 const mockData= require('./mockResult.js')
 
-function findMaxMoviesInYear () {
+function findMaxMoviesInYear ( data ) {
     const moviesInAYear = {};
-    for ( let movie of mockData ) {
-        movie.license.reduce( ( acc, currentValue,currentIndex ) => {  
-            let currentYear = currentValue;
+    for ( let movie of data ) {
+        const movieLicenseArray = movie.license.slice();
+        const movieLicenseStartYear = movieLicenseArray[0];
+        const movieLicenseEndYear = movieLicenseArray[movieLicenseArray.length - 1];
 
-            // loop through the license array till the end of the array and not consider the second index of the license array since it would be duplicate entry.
+        let currentYear = movieLicenseStartYear;
 
-            while ( currentYear <= movie.license[movie.license.length - 1] && (currentIndex!==movie.license.length-1) ) {  
-                if ( moviesInAYear[currentYear] ) {
-                    moviesInAYear[currentYear] += 1;
-                } else {
-                    moviesInAYear[currentYear] = 1;
-                }
-                currentYear += 1;
+        while ( currentYear <= movieLicenseEndYear ) {
+            if ( moviesInAYear[currentYear] ) {
+                moviesInAYear[currentYear] += 1;
+            } else {
+                moviesInAYear[currentYear] = 1;
             }
```

Comment:
The if-else can be reduced to one liner `moviesInAYear[currentYear] =  (moviesInAYear[currentYear] + 1) || 1`

Read up more here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Data_structures

Reply:
Sure, ll do that :)